### PR TITLE
Add support for `disable_escaping` parameter for select DB Engines

### PR DESCRIPTION
**Pull request by vinay-gopalan**
This PR adds support for the `disable_escaping` field to Redshift, MSSQL, Postgres and HanaDB in order to fully support templated connection URLs in the Provider.

Relates OR Closes [#4770](https://hashicorp.atlassian.net/browse/VAULT-4770)

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note: improvement
`resource/database_secret_backend_connection`: Add support for the `username`, `password` and `disable_escaping` fields to Redshift, MSSQL, Postgres and HanaDB 
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccDatabaseSecretBackendConnection_mssql'
=== RUN   TestAccDatabaseSecretBackendConnection_mssql
--- PASS: TestAccDatabaseSecretBackendConnection_mssql (8.67s)
PASS

$ make testacc TESTARGS='-run=TestAccDatabaseSecretBackendConnection_postgresql
=== RUN   TestAccDatabaseSecretBackendConnection_postgresql
--- PASS: TestAccDatabaseSecretBackendConnection_postgresql (1.69s)
PASS
```

### CODE REVIEW

**Review by benashz — COMMENTED** [review]
A few minor nits.

We also need to update the website docs.

File: vault/resource_database_secret_backend_connection.go [hunk 1]
```diff
@@ -647,6 +651,14 @@ func connectionStringResource(config *connectionStringConfig) *schema.Resource {
 		}
 	}
 
+	if config.includeDisableEscaping {
+		res.Schema["disable_escaping"] = &schema.Schema{
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "If set, disables escaping special characters for username and password",
```
[the remark this answers is not in the document]
Comment (vinay-gopalan) [reply vault/resource_database_secret_backend_connection.go]: Updated in 7de93fa

File: vault/resource_database_secret_backend_connection_test.go [hunk 2]
```diff
@@ -23,6 +23,11 @@ import (
 	"github.com/hashicorp/terraform-provider-vault/testutil"
 )
 
+type URLInfo struct {
```
[the remark this answers is not in the document]
Comment (vinay-gopalan) [reply vault/resource_database_secret_backend_connection_test.go]: Turns out I didn't end up using the `URLInfo ` struct, so I removed it in 7de93fa. Thanks for catching it!

File: website/docs/r/database_secret_backend_connection.md [hunk 3]
```diff
@@ -167,6 +167,10 @@ Exactly one of the nested blocks of configuration options must be supplied.
   docs](https://www.vaultproject.io/api-docs/secret/databases/mongodb.html#sample-payload)
   for an example.
 
+* `username` - (Optional) The root credential username used in the connection URL.
```
[the remark this answers is not in the document]
Comment (vinay-gopalan) [reply website/docs/r/database_secret_backend_connection.md:170]: @benashz good call. I created another PR to be merged to main here #1344

**Pull request merged**